src/background/adblocker.js

```diff
@@ -98,25 +105,26 @@ async function updateEngines() {
       await Promise.all(
         enabledEngines
           .filter((id) => id !== engines.CUSTOM_ENGINE)
-          .map((id) =>
-            engines.update(id).then(
-              (v) => {
-                updated = updated || v;
-              },
-              () => {},
-            ),
-          ),
+          .map(async (id) => {
+            try {
+              await engines.init(id);
+              const updatedEngine = await engines.update(id);
+              updated ||= updatedEngine;
+            } catch {
+              // Ignore errors
+            }
+          }),
       );
 
-      // Reload the main engine after all engines are updated
-      if (updated) await reloadMainEngine();
-
       // Update TrackerDB engine
       trackerdb.setup.pending && (await trackerdb.setup.pending);
       await engines.update(engines.TRACKERDB_ENGINE).catch(() => null);
 
       // Update timestamp after the engines are updated
       await store.set(Options, { filtersUpdatedAt: Date.now() });
+
+      // Reload the main engine after all engines are updated
+      if (updated) await reloadMainEngine();
```

Comment:
The reload can happen for many reasons, but the timing should only be connected to the update process. Because of that, I moved the `reloadMainEngine()` call after setting a new date. More importantly, if for some reason the `reloadMainEngine` would throw, we will not end with an endless loop of updates.